src/main/java/org/testng/collections/Lists.java

```diff
@@ -47,4 +48,23 @@ private Lists() {}
     result.addAll(l2);
     return result;
   }
+
+  /**
+   * Utility method that merges two lists by applying the provided condition.
+   * @param l1 - The first list
+   * @param l2 - The second list which is to be merged into the first list
+   * @param condition - The condition that is used to determine if an element is to be added or not.
+   * @param <T> - The generic type
+   * @return - The merged list.
+   */
+  public static <T> List<T> merge(List<T> l1, List<T> l2, BiPredicate<T, T> condition) {
```

Comment:
`public static <T> List<T> merge(List<T> list, BiPredicate<T, T> condition, List<T>... lists)`

will allow you to pass more than 2 lists (you need 3 lists somewhere)

Reply:
Yes got it. I kept it specific to solve just this problem instead of making it a more generic implementation. Will change it accordingly.